pkg/scorecard.go

```diff
@@ -102,6 +121,11 @@ func RunScorecard(ctx context.Context,
 	if err != nil || commitSHA == "" {
 		return ScorecardResult{}, err
 	}
+	defaultBranch, err := repoClient.GetDefaultBranchName()
+	if err != nil {
+		return ScorecardResult{}, err
+	}
```

Comment:
Good catch. I will return an error only if the retuned error is not `clients.ErrUnsupportedFeature`. If the errors is `clients.ErrUnsupportedFeature`, I'll set the defaultBranch to "unknown". It's no ideal because the fix will be incorrect. This has some impact on Scorecard run on PR (which we don't official support). Created https://github.com/ossf/scorecard/issues/3069 for tracking.